libraries/AP_Param/AP_Param.cpp

```diff
@@ -2388,6 +2399,39 @@ bool AP_Param::set_by_name(const char *name, float value)
     return false;
 }
 
+/*
+  get a value by name
+ */
+bool AP_Param::get(const char *name, float &value)
+{
+    enum ap_var_type vtype;
+    AP_Param *vp = find(name, &vtype);
```

Comment:
this find() call is very expensive. We need to be sure that the caller does not hold any semaphores